packages/block-library/src/button/edit.native.js

```diff
@@ -382,7 +391,7 @@ class ButtonEdit extends Component {
 							this.onToggleButtonFocus( true )
 						}
 						__unstableMobileNoFocusOnMount={ ! isSelected }
-						selectionColor={ textColor.color || '#fff' }
+						selectionColor={ this.getTextColor() }
```

Comment:
I believe that is should be `getBackgroundColor` :)

Reply:
Nope, selection color depends on textColor